protostar/protostar_toml/io/protostar_toml_writer.py

```diff
@@ -12,7 +13,6 @@
 
 
 class ProtostarTOMLWriter:
-    # pylint: disable=no-self-use
     def save(
```

Comment:
I think we path should be a constructor argument, if you want to make it a class.

Reply:
Otherwise make this method static

Reply:
I created this class because:
- `ProtostarTOMLReader` suggests the existence of this class
- classes are easier to provide as a dependency, therefore it makes testing easier 

`Path` is a configuration, not a dependency argument. It should be provided to save method. 

I don't want to make a static method that isn't pure.
Static methods have to be patched in test files, and patching is more fragile than mocking.

Reply:
>I don't want to make a static method that isn't pure.

Why? staticmethod has nothing to do with being pure. It means method does not alter object's state

What do you mean by patching is more fragile than mocking 🤔 ?

Reply:
Unpure statics have side effects. Side effects are bad because test cases aren't isolated. Therefore, these static methods have to be patched. Usually you don't have to patch pure static methods.


Patching is more fragile because you have to provide a string which can easily be invalidated during refactoring.

Reply:
>Unpure statics have side effects. Side effects are bad because test cases aren't isolated. 

I don't see this sentence being an valid argument. Every function with side effects is unpure. Being static has nothing to do with purity, static only means that method does not alter object state. 

I see the issue with patching, but I don't think we should write stupid code only because of pytest quirks. Such invalidation will be easy to spot as test will simply fail(if not it's probably a bad test).

Please change it to staticmethod and use staticmethods whenever method doesn't use object's state.

Reply:
I know that static has nothing to do with purity. The point I was trying to make
`Hard to mock stuff` + `Stuff that needs to be mocked` = hard to test code

> I see the issue with patching, but I don't think we should write stupid code only because of pytest quirks

~stupid~ -> testable
~ ~only~

I spend more time writing tests that actual code. Patching is unintuitive, fragile and sometimes doesn't work just because of naming collision.

> Please change it to staticmethod and use staticmethods whenever method doesn't use object's state.

Nope. I'd make it a function. I'll be avoiding statics at all costs.